src/internal_rep.rs

```diff
@@ -1770,6 +2127,15 @@ impl From<&FunctionArgument<'_>> for sexp::Sexp {
     }
 }
 
+// Two arguments are equal if their types are the same (including is_list_param).
+// Names and defaults may differ
+impl PartialEq for FunctionArgument<'_> {
+    fn eq(&self, other: &FunctionArgument) -> bool {
+        &self.name == "this" && &other.name == "this"
```

Comment:
Is it really needed to add this exception with `this`?

Reply:
This is because all functions internally have a "this" argument inserted as their first argument.  That argument has the type of the parent function.  Since in deriving "this" becomes a reference to the child type, but in its original context it referred to the parent type, these need to be handled as a special case.